client/my-sites/media/main.jsx

```diff
@@ -132,6 +139,9 @@ class Media extends Component {
 		MediaActions.update( site.ID, item, true );
 		resetAllImageEditorState();
 		this.setState( { currentDetail: null, editedImageItem: null, selectedItems: [] } );
+		if ( this.props.mediaId && site.slug ) {
+			page( '/media/' + site.slug );
+		}
```

Comment:
to account for `filter` this could be:

```
if ( this.props.mediaId && site.slug ) {
    let redirect = '/media/' + site.slug;
    if ( this.props.filter ) {
        redirect = '/media/' + this.props.filter + '/' + site.slug
    }
    page( redirect );
}
```